### PR TITLE
Change type of description field from String to JSONString

### PR DESCRIPTION
**Pull request by d-wysocki**
I want to merge this change because...

- Deprecate `descriptionJson`
- Change type of description field from String to JSONString
- migrate data drom description_json into description

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/graphql/page/tests/test_page.py [hunk 1]
```diff
@@ -127,19 +127,18 @@ def test_staff_query_unpublished_page(staff_api_client, page, permission_manage_
 
 CREATE_PAGE_MUTATION = """
     mutation CreatePage(
-            $slug: String, $title: String, $content: String, $pageType: ID!
-            $contentJson: JSONString, $isPublished: Boolean,
+            $slug: String, $title: String, $content: JSONString, $pageType: ID!
+            $isPublished: Boolean,
```
Comment (IKarbowiak) [inline saleor/graphql/page/tests/test_page.py:131]: Shouldn't we leave tests with both `contentJson` and `content` as we didn't drop it, but only deprecated it? The same situation in the other tests.

File: saleor/graphql/attribute/tests/queries/test_attribute_query.py [hunk 2]
```diff
@@ -453,7 +453,9 @@ def test_attributes_in_collection_query(
     other_collection = Collection.objects.create(
         name="Other Collection",
         slug="other-collection",
-        description="Description",
+        description={
+            "blocks": [{"data": {"text": "Test description."}, "type": "paragraph"}]
```
Comment (maarcingebala) [inline saleor/graphql/attribute/tests/queries/test_attribute_query.py:457]: Maybe we should have a function that would turn some text into simply editor.js content? Something like `dummy_editorjs("Test description")` that would output the above structure? Just for tests.

File: saleor/graphql/page/tests/test_page.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/graphql/page/tests/test_page.py:131]: We deprecate it in types but remove it in mutations (mutations are internal API and we will fix the dashboard accordingly).

**Comment by maarcingebala** [comment]
@d-wysocki Please update the PR name, I think "Create data migrations" it's not the right one.

**Comment by maarcingebala** [comment]
Also, let's not forget about the changelog, where it would be good to list the fields that are deprecated in this PR.

File: saleor/page/migrations/0018_migrate_content_json_into_content.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    Page = apps.get_model("page", "Page")
+    PageTranslation = apps.get_model("page", "PageTranslation")
+
+    models = [Page, PageTranslation]
+
+    for model in models:
+
+        for item in model.objects.iterator():
+            item.content = "{}"
+            item.save(update_fields=["content"])
```
Comment (fowczarek) [inline saleor/page/migrations/0018_migrate_content_json_into_content.py:19]: Maybe? 
```suggestion
    Page.objects.all().update(content="{}")
    PageTranslation.objects.all().update(content="{}")
```

File: saleor/page/migrations/0018_migrate_content_json_into_content.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    Page = apps.get_model("page", "Page")
+    PageTranslation = apps.get_model("page", "PageTranslation")
+
+    models = [Page, PageTranslation]
+
+    for model in models:
+
+        for item in model.objects.iterator():
+            item.content = item.content_json
+            item.save(update_fields=["content"])
```
Comment (fowczarek) [inline saleor/page/migrations/0018_migrate_content_json_into_content.py:32]: Maybe?
```suggestion
    Page.objects.all().update(content=F("content_json"))
    PageTranslation.objects.all().update(content=F("content_json"))
```

File: saleor/product/migrations/0138_migrate_description_json_into_description.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        Collection,
+        CollectionTranslation,
+        Product,
+        ProductTranslation,
+    ]
+
+    for model in models:
+        for item in model.objects.iterator():
+            item.description = "{}"
+            item.save(update_fields=["description"])
```
Comment (fowczarek) [inline saleor/product/migrations/0138_migrate_description_json_into_description.py:29]: Maybe?
```suggestion
        model.objects.all().update(description="{}")
```

File: saleor/product/migrations/0138_migrate_description_json_into_description.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        Collection,
+        CollectionTranslation,
+        Product,
+        ProductTranslation,
+    ]
+
+    for model in models:
+        for item in model.objects.iterator():
+            item.description = item.description_json
+            item.save(update_fields=["description"])
```
Comment (fowczarek) [inline saleor/product/migrations/0138_migrate_description_json_into_description.py:52]: Maybe?
```suggestion
        model.objects.all().update(description=F("description_json"))
```

**Review by maarcingebala — APPROVED** [review]

**Comment by maarcingebala** [comment]
Adding a `blocked` label since we need to merge it along with a dashboard PR.

**Review by fowczarek — APPROVED** [review]

**Pull request merged**